internal/database/db.go

```diff
@@ -362,6 +362,45 @@ func (dbs *SQLdb) setArchived(file FileInfo, id int64) error {
 	return nil
 }
 
+// CheckAccessionIdExists validates if an accessionID exists in the db
+func (dbs *SQLdb) CheckAccessionIDExists(accessionID, filepath, checksum string) (bool, error) {
+
+	var err error
+	var exists bool
+
+	// 3, 9, 27, 81, 243 seconds between each retry event.
+	for count := 1; count <= dbRetryTimes; count++ {
+		exists, err = dbs.checkAccessionIDExists(accessionID, filepath, checksum)
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Duration(math.Pow(3, float64(count))) * time.Second)
+	}
+
+	return exists, err
+}
+
+// checkAccessionIdExists validates if an accessionID exists in the db
+func (dbs *SQLdb) checkAccessionIDExists(accessionID, filepath, checksum string) (bool, error) {
+	dbs.checkAndReconnectIfNeeded()
+
+	db := dbs.DB
+
+	const checkIDExist = "SELECT COUNT(stable_id) FROM local_ega.files WHERE stable_id = $1 " +
+		"and inbox_path = $2 and decrypted_file_checksum = $3 and status = 'READY';"
```

Comment:
Explicitly checking for path, checksum and staus might be a mistake here, as there could exist files with different value of these fields, but the same stable_id, and they'll still cause a unique key collision. Something like `SELECT COUNT(*) FROM sda.files WHERE stable_id = $1` is likely sufficient and safer. :)